pallene/types.lua

```diff
@@ -27,6 +27,7 @@ declare_type("T", {
         "field_names", -- same order as the source type declaration
         "field_types", -- map { string => types.T }
     },
+    Module  = {},
```

Comment:
Adding this case to the `Types.T` union makes me nervous. The fake "Void" type has already caused trouble for us and this would have the same pattern.

Reply:
I think I would need to bypass some checks made by the compiler, if I were to remove this, but I can try.

Reply:
Could you clarify what you mean by "need to bypass some  checks"?

Reply:
I believe "bypass" was a poor choice of verb. I meant that I'll need to add exceptions wherever a type is expected and we have a module variable. Putting it in the `Types.T` union simplified things a little, but I do see your concern and I also agree that it shouldn't be treated as a regular type. I'll remove it from the union and handle the exceptions, np.

Reply:
The main thing that would worry me is code that might silently accept the fake module type. For example, `type.equals` might say that `module == module`. Another potential problem would be type inference; `local x = m` might infer `x` to have type `module` if `m` is a module reference.

But I agree 100% that keeping things simple is an important goal. One thing we could try is to raise an error as soon as we convert an `ast.Type` to a `types.T`. Only allow the "module" type annotation to be used for declaring a module reference in the toplevel.